syntax = "proto3";

enum OperationType {
  CREATE = 0;
  UPDATE = 1;
  DELETE = 2;
}

message Inventory {
  optional int32 id = 1;
  optional int32 product_id = 2;
  optional int32 stock_level = 3;
  optional OperationType operation = 4;
}